### PR TITLE
AkaTile

### PR DESCRIPTION
**Pull request by canuse**
issue #14

### CODE REVIEW

**Comment by canuse** [comment]
I remove Tile hash check of AkaTiles for if 5s is already in the hashlist, 0s will be added to the 5s item.
That is in [this commit](https://github.com/Ledenel/auto-white-reimu/pull/21/commits/11dfb8a7a469f5ae47f67b0122848886c26fccef)

So mulitiple Aka Tile in a single game (for example, 2 or more 0s) may cause problem in current implementation, however this won't happen in tenhou games.
Not sure if this will affect other parts of the project.

**Comment by Ledenel** [comment]
Thanks for the contribution!
Please add unit test like

https://github.com/Ledenel/auto-white-reimu/blob/master/tests/test_heuristic_pattern_waiting.py

to ensure that `AkaTile` work well with core algorithm in demo 2 (tenhou record win efficiency check)

**Comment by Ledenel** [comment]
> I remove Tile hash check of AkaTiles for if 5s is already in the hashlist, 0s will be added to the 5s item.
> That is in [this commit](https://github.com/Ledenel/auto-white-reimu/pull/21/commits/11dfb8a7a469f5ae47f67b0122848886c26fccef)
> 
> So mulitiple Aka Tile in a single game (for example, 2 or more 0s) may cause problem in current implementation, however this won't happen in tenhou games.
> Not sure if this will affect other parts of the project.

For a cleaner solution, you could add a brand new `_TILE_POOL` attribute in your `AkaTile` class like:
https://github.com/Ledenel/auto-white-reimu/blob/73deb12922a5ba29063c3f1e985185d742c74f8d/mahjong/tile/definition.py#L31

Then, use `AkaTile._TILE_POOL` in your class when possible.

File: mahjong/tile/definition.py [hunk 1]
```diff
@@ -116,3 +115,43 @@ def tile_triplet(tile: Tile) -> TileSet:
...
+
+class AkaTile(Tile):
+    _ATILE_POOL = {}
+
+    def __new__(cls, number: int, color: str):
+        tile_hash = tile_hash_num(color, number)
+        if tile_hash in AkaTile._ATILE_POOL:
+            return AkaTile._ATILE_POOL[tile_hash]
+        else:
+            tile_hash = tile_hash_num(color, number + 5)
```
Comment (Ledenel) [inline mahjong/tile/definition.py]: Do we need to reject any creation of non-5 aka tiles, to ensure no one will happened to create a **Aka 6s** by passing (1s) to this constructor? I mean, shall we `assert number == 0` at first?

File: mahjong/record/utils/value/tile.py [hunk 2]
```diff
@@ -1,14 +1,21 @@
...
 
 
 def tile_from_tenhou(index):
     """
     tenhou index is numbered as [1m,1m,1m,1m,2m,2m,...,9m,1p,...,9p,1s,...,9s,1z,...,7z,7z,7z,7z]
     numbered first 5m,5s,5p is considered as aka dora.
     """
-    color, number, _ = TENHOU_TILE_CATEGORY.category(index)
-    return Tile(number + 1, SUIT_ORDER[color])
+    if index == 16:
```
Comment (Ledenel) [inline mahjong/record/utils/value/tile.py]: the number `16,52,88` is counter-intutive at first. consider using something like `TENHOU_TILE_CATEGORY.index(('m', 5, 0))` , like below `tile_to_tenhou_range()` does.

**Comment by Ledenel** [comment]
LGTM! This could be a excellent example of how to contribute on this project.

**Pull request merged**